src/ui/game/GameScreen.js

```diff
@@ -48,17 +48,25 @@ type State = {
 };
 
 export default class GameScreen extends Component<Props, State> {
+  static getDerivedStateFromProps(props: Props, state: State) {
+    let { chatSections } = state;
+    let { game } = props;
+    let currentChatSections = getGameChatSections(game);
+
+    if (chatSections.length !== currentChatSections.length) {
+      return { ...state, chatSections: currentChatSections };
+    }
+
+    return null;
+  }
+
   state = {
     tab: "chat",
-    chatSections: getGameChatSections(this.props.game),
+    chatSections: [],
   };
 
   _chatScrollRef: ?HTMLElement;
 
-  /*  componentWillReceiveProps(nextProps: Props) {
```

Comment:
This was the original code that updated the chat. Looks like it's been commented out for a couple years now.